### DISCUSSION TITLE
Turtle parser accepts literal subjects, which is invalid RDF

### DISCUSSION THREAD

**Discussion by miguelvaara** [RDFLib/rdflib#1614] [Q&A] [answered]
While running a script including this kind of loop,

```
for s, p, o in g.triples((None, None, skos.Concept)):
  some code
```

..an accidentally added **literal as a subject** is surprisingly allowed (originally from the loaded turtle formatted file). _None_ as a subject can read types (in this case literals) outside the standard. An assumption is that it is the parser that allows non-standard subject types.

Python: 3.6.9
RDFLib: 5.0.0

**Comment by osma** [comment]
To clarify: The problem is with the Turtle parser, not the .triples method (where it surfaced). The parser seems to happily accept a literal in the subject position, even though this is not allowed in RDF.

Contents of test.ttl:

    "example" <http://example.org/p> <http://example.org/o> .

Code:

```
>>> import rdflib
RDFLib Version: 5.0.0
>>> g = rdflib.Graph()
>>> g.parse('test.ttl', format='turtle')
<Graph identifier=Nec6e885a6e8e4d36bc35621f099c2b49 (<class 'rdflib.graph.Graph'>)>
>>> for triple in g.triples((None, None, None)):
...     print(triple)
... 
(rdflib.term.Literal('example'), rdflib.term.URIRef('http://example.org/p'), rdflib.term.URIRef('http://example.org/o'))
```

The N-Triples parser fails on the same file.

**Comment by osma** [comment, marked as answer]
Related to #895 - it appears that rdflib is not very strict about RDF and in some cases (but not always) supports [generalized RDF datasets](https://www.w3.org/TR/rdf11-concepts/#dfn-generalized-rdf-dataset) where literals can be in the subject position.
